model/task_lifecycle.go

```diff
@@ -521,8 +521,11 @@ func UpdateBlockedDependencies(t *task.Task) error {
 		if err = UpdateBlockedDependencies(&dependentTask); err != nil {
 			return errors.WithStack(err)
 		}
-		if err = build.SetCachedTaskBlocked(dependentTask.BuildId, dependentTask.Id, dependentTask.Blocked()); err != nil {
-			return errors.Wrap(err, "error marking cached task as blocked")
+
+		if !dependentTask.IsPartOfDisplay() { // execution tasks not cached in build so we should skip
```

Comment:
if it's part of a display task and it's blocked won't the display task be blocked?

Reply:
Shouldn't the display task be blocked if the execution task is blocked?

Reply:
I think so.
It looks to me like it will not be marked blocked in the build cache if we don't do anything with the build cache when an execution task completes.

Reply:
I was wrong: we will call it on the display task too